### PR TITLE
Add modifier `RemoveFromVendor::groups()` and add ability specify groups in `RemoveFromVendor::keys()`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #92 #93

### CODE REVIEW

File: src/Modifier/RemoveGroupsFromVendor.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Config\Modifier;
+
+use function is_array;
+
+/**
+ * @see RemoveFromVendor::groups()
+ */
+final class RemoveGroupsFromVendor
```
Comment (devanych) [inline src/Modifier/RemoveGroupsFromVendor.php:12]: Maybe mark as internal?

File: src/Modifier/RemoveKeysFromVendor.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Config\Modifier;
+
+use InvalidArgumentException;
+
+/**
+ * @see RemoveFromVendor::keys()
+ */
+final class RemoveKeysFromVendor
```
Comment (devanych) [inline src/Modifier/RemoveKeysFromVendor.php:12]: Maybe mark as internal?

**Review by devanych — APPROVED** [review]

File: src/Modifier/RemoveGroupsFromVendor.php (hunk 1 above)
Comment (vjik) [reply src/Modifier/RemoveGroupsFromVendor.php:12]: `RemoveFromVendor::groups()` returned this to userland... I do not know if it is necessary to do internal.

File: src/Modifier/RemoveKeysFromVendor.php (hunk 2 above)
Comment (vjik) [reply src/Modifier/RemoveKeysFromVendor.php:12]: `RemoveFromVendor::keys()` returned this to userland... I do not know if it is necessary to do internal.

Comment (devanych) [reply src/Modifier/RemoveKeysFromVendor.php:12]: OK

File: src/Modifier/RemoveGroupsFromVendor.php (hunk 1 above)
Comment (devanych) [reply src/Modifier/RemoveGroupsFromVendor.php:12]: OK

File: src/Merger.php [hunk 3]
```diff
@@ -284,6 +297,30 @@ private function setValue(Context $context, array $keyPath, array &$array, strin
         return true;
     }
 
+    /**
+     * @psalm-param non-empty-array<array-key, string> $keyPath
+     */
+    private function isRemoveKeyFromVendor(Context $context, array $keyPath): bool
```
Comment (samdark) [inline src/Merger.php:303]:
```suggestion
    private function shouldRemoveKeyFromVendor(Context $context, array $keyPath): bool
```

**Review by samdark — APPROVED** [review]

**Pull request merged**